projects/lib/src/board/racingkingsboard.cpp

```diff
@@ -0,0 +1,132 @@
+/*
+    This file is part of Cute Chess.
+
+    Cute Chess is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Cute Chess is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Cute Chess.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "racingkingsboard.h"
+#include "westernzobrist.h"
+
+namespace Chess {
+
+RacingKingsBoard::RacingKingsBoard()
+	: WesternBoard(new WesternZobrist())
+{
+}
+
+Board* RacingKingsBoard::copy() const
+{
+	return new RacingKingsBoard(*this);
+}
+
+QString RacingKingsBoard::variant() const
+{
+	return "racingkings";
+}
+
+QString RacingKingsBoard::defaultFenString() const
+{
+	return "8/8/8/8/8/8/krbnNBRK/qrbnNBRQ w - - 0 1";
+}
+
+bool RacingKingsBoard::vIsLegalMove(const Move& move)
```

Comment:
It doesn't look like you need to reimplement `vIsLegalMove`. `Board::vIsLegalMove` already looks like this:

```
	makeMove(move);
	bool isLegal = isLegalPosition();
	undoMove();

	return isLegal;
```

So you could just reimplement `WesternBoard::isLegalPosition` like this:

```
bool RacingKingsBoard::isLegalPosition()
{
	if (inCheck(sideToMove()))
		return false;
	return WesternBoard::isLegalPosition();
}
```

I didn't test it so it's possible I missed something.